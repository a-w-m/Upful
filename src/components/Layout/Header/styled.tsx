import React from "react"
import styled from "styled-components"
import device from "../mediaQuery"
import { Link } from "gatsby"

export const HeaderContainer = styled.header`
  
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: flex-end;
  padding: 2rem 2rem 0rem 2rem;

  @media ${device.tablet}{
    justify-content: flex-end;
    gap: 2rem;
    grid-area: header;

    /*remove logo from header at tablet width and larger */
    & > a:nth-child(2){
      display: none;
    }

  }

  @media ${device.laptop}{
    padding: 2rem 4rem 0rem 4rem;
  }
`

export const HamburgerNavContainer = styled.div<{open: boolean}>`
width: ${({open})=> open? "100%": "auto"};
align-self: flex-start;
@media ${device.laptop}{
  order: 1;
  flex-basis: 100%;
}

`

export const LogoContainer = styled.div`
  max-width: 960;
  flex-basis: 100%;
  display: flex;
  justify-content: center;
`

export const LogoWrapper = styled.div`
  flex-basis: 25%;

  @media ${device.laptop} {
    flex-basis: 15%;
  }
`

export const LinkWrapper = styled.h1`
  margin: 0;
  font-family: var(--main-font);
`

export const TitleLink = styled(props => <Link {...props} />)`
  color: #000;
  text-decoration: none;
`

export const CartContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  align-self: flex-start;
`
export const Checkout = styled.button`
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  line-height: 1.5;
  text-decoration: none;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  flex-direction: row;
  -webkit-box-align: center;
  align-items: center;
  font-size: 1rem;
  color: #666;
  font-family: Open Sans, sans-serif;
  font-family: var(--fontFamily);
  font-weight: 700;
`

export const Count = styled.span`
  align-self: flex-end;
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 2rem;
`
