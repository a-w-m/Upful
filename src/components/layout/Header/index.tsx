import PropTypes from "prop-types"
import React from "react"
const { useState, useRef, useContext } = React

import Hamburger from "../Hamburger"
import Nav from "../Nav"
import { Link } from "gatsby"

import { StaticImage } from "gatsby-plugin-image"
import { SnipcartContext } from "gatsby-plugin-snipcart-advanced/context.js"

import {
  HeaderContainer,
  LogoContainer,
  LogoWrapper,
  CartContainer,
  Checkout,
  Count,
  HamburgerNavContainer
  
} from "./styled"
import { Cart } from "../../Icons"
import useOnClickOutside from "../../../hooks/useOnClickOutside"

interface HeaderProps {
  siteTitle: String
}

const Header: React.FC<HeaderProps> = ({ siteTitle }) => {
  const node = useRef(null)
  const [open, setOpen] = useState(false)
  useOnClickOutside(node, () => setOpen(false))
  const { state } = useContext(SnipcartContext)
  const { cartQuantity } = state

  return (
    <HeaderContainer>
      <HamburgerNavContainer ref={node} open ={open}>
        <Hamburger open={open} setOpen={setOpen} />
        <Nav open={open} />
      </HamburgerNavContainer>
      <CartContainer>
        <Checkout className="snipcart-checkout">
          <Cart />
        </Checkout>
        <Count>{cartQuantity}</Count>
      </CartContainer>
      <LogoContainer>
        <LogoWrapper>
          <Link to="/">
            <StaticImage
              src="../../../images/upful-gold-frame-logo.png"
              alt="logo"
              style={{ borderRadius: "100%" }}
              width={250}
            />
          </Link>
        </LogoWrapper>
      </LogoContainer>
    </HeaderContainer>
  )
}

Header.propTypes = {
  siteTitle: PropTypes.string.isRequired,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
