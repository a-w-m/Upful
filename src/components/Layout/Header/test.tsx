import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import Header from ".."

describe("Header component", () => {
  test("matches snapshot", () => {
    const { container } = render(<Header children="" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("renders Header with correct site title", () => {
    render(<Header children="" />)

    expect(screen.getByText("StoreFront")).toBeInTheDocument()
  })

  // test("displays toggle menu on small screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.getByAltText("toggle-menu")).toBeInTheDocument()

  // })

  // test("does not display toggle menu on large screens", ()=>{
  //   render(<Header children="" />)

  //   expect(screen.queryAllByAltText("toggle-menu")).toBeNull()

  // })
})

// describe("click outside toggle menu", () => {
//   test("Nav remains visible when clicking inside component ", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.click(Nav)
//     expect(Nav).toBeVisible()
//   })

//   test("Nav is not visible when clicking outside component", () => {
//     render(<Header children="" />)
//     const Button = screen.getByRole("button")
//     const Nav = screen.getAllByRole("navigation", { hidden: true })[0]
//     fireEvent.click(Button)
//     fireEvent.mouseDown(Button)
//     expect(Nav).not.toBeVisible()
//   })
// })
