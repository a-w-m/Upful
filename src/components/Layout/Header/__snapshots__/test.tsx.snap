// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  class="sc-jSMfEi dhsDWM"
>
  <div
    class="sc-gKXOVf bRgbNH"
  >
    <div
      class="sc-bczRLJ iKPXgD"
    >
      <button
        aria-controls="menu"
        aria-expanded="false"
        aria-label="toggle navigation"
        class="sc-gsnTZi jUeZLU"
        type="button"
      >
        <span />
        <span />
        <span />
      </button>
    </div>
    <nav
      aria-label="Main"
      class="sc-dkzDqf ckAmYP"
      id="menu"
    >
      <a
        class="sc-eCYdqJ kRVJaY"
        href="/"
      >
        <div
          class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
          data-gatsby-image-wrapper=""
          style="position: relative; overflow: hidden; display: inline-block; vertical-align: top; border-radius: 100%;"
        >
          <div
            style="max-width: 250px; display: block;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;charset=utf-8,%3Csvg height='250' width='250' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
              style="max-width: 100%; display: block; position: static;"
            />
          </div>
          <div
            aria-hidden="true"
            data-placeholder-image=""
            style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
          />
          <picture>
            <source
              data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/74c72/upful-gold-frame-logo.webp 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/d66e1/upful-gold-frame-logo.webp 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/e7160/upful-gold-frame-logo.webp 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/5f169/upful-gold-frame-logo.webp 500w"
              sizes="(min-width: 250px) 250px, 100vw"
              type="image/webp"
            />
            <img
              alt="logo"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png"
              data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/6a639/upful-gold-frame-logo.png 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/2fd20/upful-gold-frame-logo.png 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/82c11/upful-gold-frame-logo.png 500w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 250px) 250px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
          </picture>
          <noscript />
          <script
            type="module"
          >
            const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
          </script>
        </div>
      </a>
      <a
        class="sc-hKMtZM cdOabJ"
        href="/women/"
      >
        WOMEN
      </a>
      <a
        class="sc-hKMtZM cdOabJ"
        href="/kids/"
      >
        KIDS
      </a>
    </nav>
  </div>
  <a
    class="sc-eCYdqJ kRVJaY"
    href="/"
  >
    <div
      class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
      data-gatsby-image-wrapper=""
      style="position: relative; overflow: hidden; display: inline-block; vertical-align: top; border-radius: 100%;"
    >
      <div
        style="max-width: 250px; display: block;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;charset=utf-8,%3Csvg height='250' width='250' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
          style="max-width: 100%; display: block; position: static;"
        />
      </div>
      <div
        aria-hidden="true"
        data-placeholder-image=""
        style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
      />
      <picture>
        <source
          data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/74c72/upful-gold-frame-logo.webp 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/d66e1/upful-gold-frame-logo.webp 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/e7160/upful-gold-frame-logo.webp 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/5f169/upful-gold-frame-logo.webp 500w"
          sizes="(min-width: 250px) 250px, 100vw"
          type="image/webp"
        />
        <img
          alt="logo"
          data-gatsby-image-ssr=""
          data-main-image=""
          data-src="/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png"
          data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/6a639/upful-gold-frame-logo.png 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/2fd20/upful-gold-frame-logo.png 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/82c11/upful-gold-frame-logo.png 500w"
          decoding="async"
          loading="lazy"
          sizes="(min-width: 250px) 250px, 100vw"
          style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
        />
      </picture>
      <noscript />
      <script
        type="module"
      >
        const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
      </script>
    </div>
  </a>
  <div
    class="sc-kDDrLX fsOTBx"
  >
    <button
      aria-label="snipcart-checkout"
      class="sc-iqcoie ingRrx snipcart-checkout"
      type="button"
    >
      <svg
        class="snipcart-cart-header__icon   snipcart__icon"
        fill="none"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M51.714 20.47L55 60H9l3.286-39.53h9.857v-6.588C22.143 8.424 26.556 4 32 4c5.444 0 9.857 4.424 9.857 9.882v6.589h9.857zM25.43 13.883v16.47h-3.286v-6.587h-6.834l-2.737 32.94h38.856l-2.737-32.94h-6.834v6.588h-3.286v-16.47c0-3.634-2.947-6.589-6.571-6.589-3.624 0-6.571 2.955-6.571 6.588zm3.285 9.883V20.47h6.572v3.294h-6.572z"
          fill="#81550d"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <span
      class="sc-crXcEl iMYtNW"
    >
      0
    </span>
  </div>
</header>
`;
